org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/XMLRename.java

```diff
@@ -93,13 +98,45 @@ public WorkspaceEdit doRename(DOMDocument xmlDocument, Position position, String
 			}
 		}
 
-		textEdits.addAll(getRenameTextEdits(xmlDocument, node, position, newText));
+		cancelChecker.checkCanceled();
+		mergeOrAddTextEdits(xmlDocument, getRenameTextEdits(xmlDocument, node, position, newText), documentChanges);
 
 		cancelChecker.checkCanceled();
-		
-		return createWorkspaceEdit(xmlDocument.getDocumentURI(), textEdits);
+		return createWorkspaceEdit(documentChanges);
 	}
 
+	/**
+	 * Merges the given text edits into an existing text document edits or creates and adds new 
+	 * text document edits for a given list of text edits. 
+	 * 
+	 * @param document XML Document being edited
+	 * @param textEdits A list of Text edits to add or merge
+	 * @param documentChanges Target list of either text document edits or rename operations
+	 */
+	public static void mergeOrAddTextEdits(DOMDocument document, List<TextEdit> textEdits,
```

Comment:
Pease don't provide static method on XMLRename. Please remove public from class XMLRename (ex: see XMLReference).

I suggest that you add this method in https://github.com/eclipse/lemminx/blob/main/org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/format/TextEditUtils.java but please move this class in the `or.eclipse.lemminx.utils` package.

Reply:
Agree.